opm/simulators/wells/BlackoilWellModel_impl.hpp

```diff
@@ -1243,7 +1247,140 @@ namespace Opm {
         return {more_network_update, well_group_control_changed};
     }
 
+    // This function is to be used for well groups in an extended network that act as a subsea manifold
+    // The wells of such group should have a common THP and total phase rate(s) obeying (if possible) 
+    // the well group constraint set by GCONPROD
+    template <typename TypeTag>
+    void
+    BlackoilWellModel<TypeTag>::
+    computeWellGroupThp(const double dt, DeferredLogger& local_deferredLogger)
+    {
+        const int reportStepIdx = this->simulator_.episodeIndex();
+        const auto& network = this->schedule()[reportStepIdx].network();
+        const auto& balance = this->schedule()[reportStepIdx].network_balance();
+        const Scalar thp_tolerance = balance.thp_tolerance();
+
+        if (!network.active()) {
+            return;
+        }
+
+        auto& well_state = this->wellState();
+        auto& group_state = this->groupState();
+
+        for (const std::string& nodeName : network.node_names()) {
+            const bool has_choke = network.node(nodeName).as_choke();
+            if (has_choke) {
+                const auto& summary_state = this->simulator_.vanguard().summaryState();
+                const Group& group = this->schedule().getGroup(nodeName, reportStepIdx);
+                const auto ctrl = group.productionControls(summary_state);
+                const auto cmode = ctrl.cmode;
+                const auto pu = this->phase_usage_;
+                //TODO: Auto choke combined with RESV control is not supported
+                std::vector<Scalar> resv_coeff(pu.num_phases, 1.0);
+                Scalar gratTargetFromSales = 0.0; 
+                if (group_state.has_grat_sales_target(group.name()))
+                    gratTargetFromSales = group_state.grat_sales_target(group.name());
+
+                WGHelpers::TargetCalculator tcalc(cmode, pu, resv_coeff,
+                                                  gratTargetFromSales, nodeName, group_state,
+                                                  group.has_gpmaint_control(cmode));
+                const Scalar orig_target = tcalc.groupTarget(ctrl, local_deferredLogger);
+
+                auto mismatch = [&] (auto well_group_thp) {
```

Comment:
The name `well_group_thp` probably shadows another `well_group_thp` defined later.

Reply:
It is renamed to `group_thp`